### PR TITLE
#1 started validator file

### PR DESCRIPTION
**Pull request by rebecacalvoquintero**
#1 
created validator.js, checks email, password, confirmpassword

### CODE REVIEW

File: src/controllers/validate.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    throw new TypeError('This is not a name');
+  }
+  if (/\W/.test(name)) {
+    throw new TypeError('Name must contain alphanumeric characters');
+  }
+  if (name.length <= 1) {
+    throw new TypeError('Please enter your fullname');
+  }
+  if (name.length > 20) {
+    throw new TypeError('You have exceeded our maximum amount of characters');
```
Comment (dangerdak) [inline src/controllers/validate.js]: Could include max character length in error message :heart:

File: src/controllers/validate.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+const validatePassword = (password) => {
+  if(!password) {
+    throw new TypeError('Please insert a password');
+  }
+  if (typeof password !== 'string') {
+    throw new TypeError('This is not a valid password');
+  }
+  if (password.length < 8) {
```
Comment (dangerdak) [inline src/controllers/validate.js]: Maybe we should include length restriction in error message

File: src/controllers/validate.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    throw new TypeError('Your password is too short');
+  }
+  const passwordStrengthRegex = ('^(?=.[a-z])(?=.[A-Z])(?=.[0-9])(?=.[!@#\$%\^&*])');
+  if(!passwordStrengthRegex.test(password)){
+    throw new TypeError('Your password needs to contain at least one uppercase, one lowercase, one special character and one number');
```
Comment (dangerdak) [inline src/controllers/validate.js]: Not sure I like [password restrictions](https://arstechnica.com/information-technology/2013/06/password-complexity-rules-more-annoying-less-effective-than-length-ones/) like this, what does everyone else think?

Comment (Jen-Harris) [reply src/controllers/validate.js]: Thanks for the tips @dangerdak. The changes have been made 👍

File: src/controllers/passwordModule.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+'use strict';
+const crypto = require('crypto');
+const env2 = require('./config.env');
```
Comment (dangerdak) [inline src/controllers/passwordModule.js]: I think this line should be `require('env2')('config.env')`

Comment (Jen-Harris) [reply src/controllers/passwordModule.js]: 👍 done

File: config.env [hunk 5]
```diff
@@ -0,0 +1 @@
+SECRET = jenssecretforstudentprojectthisisit
```
Comment (polyccon) [inline config.env]: This shouldn't be on github :-*

Comment (Jen-Harris) [reply config.env]: :scream:
deleted file. Will update .gitignore and secret on next PR

**Pull request merged**